examples/shouldi/tests/binaries.py

```diff
@@ -3,77 +3,101 @@
 
 downloads = pathlib.Path(__file__).parent / "downloads"
 
+loop = asyncio.get_event_loop()
 # JavaScript
-cached_node = cached_download_unpack_archive(
-    "https://nodejs.org/dist/v14.2.0/node-v14.2.0-linux-x64.tar.xz",
-    downloads / "node.tar.gz",
-    downloads / "node-download",
-    "fa2a9dfa4d0f99a0cc3ee6691518c026887677a0d565b12ebdcf9d78341db2066427c9970c41cbf72776a370bbb42729",
+cached_node = loop.run_until_complete(
```

Comment:
We had talk at one point (a while ago) about how this file is referenced in the workflow. If the file changes, the cache directory of `downloads` is invalidated. With the way you have it now, these downloads run on import, we need to make it so that the arguments are stored in this file, and the functions they previously decorated call the download function using the arguments stored in this file.

Reply:
@pdxjohnny I've made some changes and I hope that's what was expected. However, the shouldi tests broke and I'm not sure why as the logs are ambiguous. The following error pops up in the logs:
```
 ======================================================================
ERROR: test_run (tests.test_dependency_check.TestRunDependencyCheckOp)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/dffml/dffml/dffml/util/asynctestcase.py", line 69, in run_it
    result = self.loop.run_until_complete(coro(*args, **kwargs))
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
    return future.result()
  File "/home/runner/work/dffml/dffml/examples/shouldi/tests/test_dependency_check.py", line 35, in test_run
    results = await run_dependency_check(str(rxjava / "RxJava-2.2.16"))
  File "/home/runner/work/dffml/dffml/examples/shouldi/shouldi/java/dependency_check.py", line 66, in run_dependency_check
    raise DependencyCheckError(stderr.decode())
shouldi.java.dependency_check.DependencyCheckError

----------------------------------------------------------------------
```

Tried reviewing my own code but found nothing as the error doesn't say much about what is at fault. 
Made the following debug statements in `run_dependency_check`:
```
                logging.debug(stderr)
                logging.debug(pkg)
                logging.debug(proc.returncode)
                logging.debug(stderr.decode())
```
which gave the following in logs:
```
DEBUG:root:b''
DEBUG:root:/home/runner/work/dffml/dffml/examples/shouldi/tests/downloads/RxJava-download/RxJava-2.2.16
DEBUG:root:243
DEBUG:root:
ERROR
```
note that `stderr.decode()` gives a blank line.